edgelet/edgelet-core/src/network.rs

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+use crate::DEFAULT_NETWORKID;
+use serde_derive::{Deserialize, Serialize};
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct Network {
+    name: String,
+
+    #[serde(rename = "ipv6", skip_serializing_if = "Option::is_none")]
+    ipv6: Option<bool>,
+
+    #[serde(rename = "ipam", skip_serializing_if = "Option::is_none")]
+    ipam: Option<Vec<Ipam>>,
+}
+
+impl Network {
+    pub fn new(name: String) -> Self {
+        Network {
+            name,
+            ipv6: None,
+            ipam: None,
+        }
+    }
+
+    pub fn name(&self) -> &str {
+        &self.name
+    }
+
+    pub fn with_name(mut self, name: String) -> Self {
+        self.name = name;
+        self
+    }
+
+    pub fn ipv6(&self) -> Option<&bool> {
```

Comment:
Return `Option<bool>` since it is `Copy`, ie `fn ipv6(&self) -> Option<bool> { self.ipv6 }`